### PR TITLE
added cluster migration worker

### PR DESCRIPTION
**Pull request by yujunglo**
@sbaogang 
Added 1 worker and 2 boolean in hostDAO
1. can_retire (default=0) indicated this host is scheduled for replacement/migration
2. can_replace (default=1); if can_replace=1, terminate host directly; otherwise, terminate and then reduce the cluster capacity.

Replace Cluster button:
screen shot 2016-04-26 at 2 00 45 pm

screen shot 2016-04-26 at 2 11 55 pm

If found other non-cmp groups,
screen shot 2016-04-26 at 2 02 05 pm

Show messages in env_landing page if env is scheduled for cluster replacement.
screen shot 2016-04-26 at 2 02 36 pm

### CODE REVIEW

**Comment by sbaogang** [comment]
Let us not use RED for warnings or call for attention - use yellow instead

File: deploy-board/deploy_board/templates/clusters/clusters.html [hunk 1]
```diff
@@ -30,12 +30,11 @@ <h4 class="panel-title pull-left">Cluster Manager</h4>
             data-toggle="modal" title="Delete Cluster">
             <span class="glyphicon glyphicon-remove-sign"></span> Delete cluster
             </button>
-            <!--- TODO To be implemented
-            <button class="deployToolTip btn btn-default btn-block" data-target="#replaceFleet"
-            data-toggle="modal" title="Replace fleet">
+
+            <button class="deployToolTip btn btn-default btn-block" data-target="#replaceCluster"
+            data-toggle="modal" title="Replace Cluster">
```
Comment (sbaogang) [inline deploy-board/deploy_board/templates/clusters/clusters.html]: Terminate and relaunch all the hosts in the cluster with new configurations.

File: deploy-board/deploy_board/templates/clusters/clusters.html [hunk 2]
```diff
@@ -82,4 +81,46 @@ <h4 class="modal-title">Delete Cluster Confirm</h4>
...
+    <div class="modal-dialog modal-md">
+        <div class="modal-content">
+            <form id="replaceClusterFormId" class="form-horizontal" role="form" method="post" action="/env/{{ env.envName }}/{{ env.stageName }}/config/replace_cluster/">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+                    <h4 class="modal-title">Cluster Replacement Confirm</h4>
+                </div>
+                <div class="modal-body" id="newMetricModal">
+                <div class="alert alert-danger">
+                    Are you sure to <strong>Replace</strong> all hosts in cluster?<br>
```
Comment (sbaogang) [inline deploy-board/deploy_board/templates/clusters/clusters.html]: ..in this cluster?

**Comment by yujunglo** [comment]
superseded by #144

**Pull request closed without merging**